### DISCUSSION TITLE
为什么不能设置导出的图片素材质量？

### DISCUSSION THREAD

**Discussion by longqi87** [Tencent/libpag#2301] [Q&A] [answered]
感觉pag包的图片被压缩有点糊了，渐变色也出现断层，能不能调节素材图片的质量哇？

**Comment by kevingpqi123** [comment]
反馈建议将背景、期望实现的效果、怎么实现的、哪里不符合预期、复现场景如 AE 工程提供齐全，就描述了一个现象，怎么会知道你遇到的问题是啥？这不是在浪费彼此的时间吗？

**Reply by longqi87** [reply]
搞笑哦。我只是作为用户反馈一下问题，你这种是什么语气？我只是作为用户提个建议，让用户能否有一个调节图片质量的功能，因为导出来的包，里面的素材肯定被压缩过，图片就必然会出现一点糊或者色差，这问题需要什么复现场景吗？素材有压缩就肯定会出现色差或者模糊啊。服

**Reply by kevingpqi123** [reply]
这样说吧，我不知道你遇到的具体问题是啥，对于 pag 而言，原则上是不会存在这种问题的，但图片导出质量发生变化可以是多方面的原因，我不太了解你这边是 AE 工程对图片做了 Transform 变化还是其它的，不知道你反馈的问题是啥，我怎么回答你？我们需要的是场景，不是你这边的分析，因为原则上导出层面默认的图片设置已经够大部分场景使用了
正确的提问姿势：
将你遇到的 AE 工程做为附件发出来，指出哪个图层的表现和导出不符合预期，你将具体的问题发出来，我们看后不就知道具体原因是啥了吗

**Comment by longqi87** [comment]
[测试包.zip](https://github.com/Tencent/libpag/files/15384286/default.zip)
AE文件， 3秒的视频 2080*1136 ，2k的分辨率了，但是pag包也仅有836kb，
![image](https://github.com/Tencent/libpag/assets/31119502/7e50bd63-70ff-47e5-815b-c04a3b0497fe)
这个压缩率


![问题图示](https://github.com/Tencent/libpag/assets/31119502/e2f9a9eb-8887-44f1-ad79-368f8597d025)
问题截图 ，难看的渐变断层（底图没做任何动画）


<img width="980" alt="render-index-geshi2" src="https://github.com/Tencent/libpag/assets/31119502/29d73cec-ada9-47fd-9942-6091ba48eaa7">
原高清素材，无渐变断层

**Reply by kevingpqi123** [reply, marked as answer]
你这边是将 AE 工程通过 BMP 预合成的方式导出，不是单纯的图片图层问题
1、从性能的角度而言，你导出的是位图序列帧，而不是视频序列帧，对于 BMP 预合成的大尺寸导出，优先推荐使用视频序列帧，在终端视频的解码可以使用硬解，但位图序列只能使用 CPU，因此视频的解码性能要比图片序列好很多；
2、针对你提到的导出分辨率问题，默认 BMP 预合成的最大短边尺寸是 720，虽然尺寸大画面是会更清晰，但终端解码压力会很多，会出现卡顿的问题，可以看到你导出的文件：
<img width="852" alt="企业微信截图_b2f3698c-9e5c-4eb1-b56d-c45938e83dfe" src="https://github.com/Tencent/libpag/assets/16609221/6170740e-0989-4756-837f-09160add8083">

但是，如果你特别追求清晰度，我们也提供了修改入口，可以看下面的文档：
https://pag.art/docs/plugin-config.html

<img width="911" alt="企业微信截图_fa2b4776-d28c-4450-bf03-f83a1952d8d6" src="https://github.com/Tencent/libpag/assets/16609221/4e464b11-1a3c-46ce-8340-1874ba6664ab">

这里需要注意的是，这里的设置是全局的，修改后会影响所有后续文件的导出，如果仅仅是一次性的需求，注意导出后还原这里的配置。
